### PR TITLE
Bs poc pr

### PR DESCRIPTION
**Pull request by vivek**
# Description

The fixes in the PR are broadly in these categories:

* JWT related changes (jwt is still disabled till we have artifact and log link work with it, there is another ticket for that). There are tons of fixes/enhancements in JWT that have been tested to verify
* Remove singleton hardcoding from backend code
* Dependency changes to narrow down to whats needed (look for pom.xml files)
* Java dependencies moved from blueocean-dashboard over to pipeline-api-impl. It's purely UI module with this change
* Other miscellaneous fixes

In front end:
* blueocean-core-js - jwt.js and other minor fixes
* blueocean-dashboard.js - minor fixes

@imeredith PTAL specially at UI changes and of course JWT and others as you are familiar at that part of the code.

See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by michaelneale** [comment]
@vivek does this address some of the tickets listed in: https://issues.jenkins-ci.org/issues/?jql=labels%20%3D%20cloudbees-internal-steel

Also - what areas should be looked at closely for reviewing? There is a fair bit to take in here.

**Comment by vivek** [comment]
@michaelneale It doesn't address any ticket over there, but it lays out foundations so that we can implement things like removing hardcoded organization from UI. I have listed specific areas that needs review under description: see https://github.com/jenkinsci/blueocean-plugin/pull/956#issue-220783944.

**Comment by michaelneale** [comment]
ok, it's a big one, will need some time for Ivan and Keith (not sure if they will have time this week).

**Comment by vivek** [comment]
@michaelneale Ok thanks, in the meantime, I will use this branch as base for other tech-debt ticket.

**Comment by michaelneale** [comment]
@vivek it is failing the ATH (perhaps this is off an older master branch?) so that warrants looking into in the meantime.

**Comment by vivek** [comment]
Right, will look in to failing ATH.

**Comment by cliffmeyers** [comment]
UI changes look okay to me, although I think there are more places where "jenkins" is hard-coded as the organization name. Is the intent of this PR to fix all of them, or just specific cases? @vivek

**Comment by vivek** [comment]
@cliffmeyers doesn't fix them all. There is separate ticket for that.

**Comment by vivek** [comment]
@michaelneale running ATH locally, only one test failed `multibranch/commitMessages`.

```
Timed out while waiting for element <.activity-table tr#commitMessages-1 svg circle.success> to be present for 20000 milliseconds.  - expected "found" but got: "not found"
```
Looking at ATH jerkins, commitMessages job looks fine and this test shouldn't have failed probably not legit. I am triggering another ATH run for this branch. If you get chance will be great if you could try it locally.

**Comment by cliffmeyers** [comment]
@vivek yeah, `commitMessages` is notoriously flaky. Very likely a spurious failure.

**Comment by michaelneale** [comment]
I'll take a look at that test today - generally its flakyness is hidden... it is odd if it is consistently failing, but may just be "luck"

**Comment by vivek** [comment]
@michaelneale I think I fixed the bug causing ATH failure. blueocean-events was supposed to be added in aggregator pom to get all sse dependencies as it was removed from dashboard. Hope ATH passes now, it does locally now include `multibranch.commitMessage`.

**Comment by michaelneale** [comment]
@vivek it passes now - that pom is the parent pom not the aggregator though (but I guess same effect). Just very very nervous about pom changes like this...

**Comment by vivek** [comment]
@michaelneale Yeah, hopefully we don't have to touch them again. It was needed to clean up accumulated tech debt. And it's aggregator pom, 'blueocean/pom.xml'.

**Comment by michaelneale** [comment]
ok, seems good to me. @vivek @kzantow would this dependency juggling have any impact on the deps as specified in: https://github.com/jenkinsci/blueocean-pipeline-editor-plugin - ie at development time? 

This works fine and passes ATH, I am just nervous, but can't quite say why. I think if @kzantow says it is ok - good with me. Other changes seem fine.

**Comment by michaelneale** [comment]
yes I tried running the editor pointing to this snapshot, it seemed "odd" - maybe it is missing some dependency, so I think some more manual testing is.

File: blueocean-jwt/src/main/java/io/jenkins/blueocean/auth/jwt/JwtAuthenticationService.java [hunk 1]
```diff
@@ -14,7 +14,7 @@
  *
  * @author Vivek Pandey
  */
-public abstract class JwtAuthenticationService implements RootAction, ExtensionPoint{
+public abstract class JwtAuthenticationService implements UnprotectedRootAction, ExtensionPoint{
```
Comment (michaelneale) [inline blueocean-jwt/src/main/java/io/jenkins/blueocean/auth/jwt/JwtAuthenticationService.java]: This looks risky... but I can understand why... is this covered by a test?

**Comment by michaelneale** [comment]
Looked at this some more - it seems to work well. Would like @kzantow to confirm it looks ok - and there was one comment above about the unprotected root action, but otherwise, good.

File: blueocean-jwt/src/main/java/io/jenkins/blueocean/auth/jwt/JwtAuthenticationService.java (hunk 1 above)
Comment (vivek) [inline blueocean-jwt/src/main/java/io/jenkins/blueocean/auth/jwt/JwtAuthenticationService.java]: Yeah, there is test that calls token generation and wks service. See [here](https://github.com/jenkinsci/blueocean-plugin/blob/bs-poc-pr/blueocean-jwt/src/test/java/io/jenkins/blueocean/auth/jwt/impl/JwtAuthenticationServiceImplTest.java#L29).

**Comment by vivek** [comment]
ATH passes with latest changes: https://ci.blueocean.io/blue/organizations/jenkins/ATH-Jenkinsfile/detail/master/2116/pipeline.

**Pull request merged**